apm-sniffer/apm-sdk-plugin/spring-plugins/mvc-annotation-commons/src/main/java/org/apache/skywalking/apm/plugin/spring/mvc/commons/interceptor/AbstractMethodInterceptor.java

```diff
@@ -208,4 +218,32 @@ private void collectHttpParam(HttpServletRequest request, AbstractSpan span) {
             Tags.HTTP.PARAMS.set(span, tagValue);
         }
     }
+
+    private boolean shouldCollectHeader() {
+        List<String> includeHeaders = SpringMVCPluginConfig.Plugin.Http.INCLUDE_HTTP_HEADERS;
+        return !CollectionUtil.isEmpty(includeHeaders) ;
+    }
+
+    private void collectHttpHeaders(HttpServletRequest request, AbstractSpan span) {
+        final Enumeration<String> headerNames =  request.getHeaderNames();
+        if (headerNames == null) {
+            return;
+        }
+        final List<String> headersList = new LinkedList<>();
+        SpringMVCPluginConfig.Plugin.Http.INCLUDE_HTTP_HEADERS.stream().filter(headerName -> request.getHeaders(headerName) != null).forEach(headerName -> {
+            Enumeration<String> headerValues = request.getHeaders(headerName);
+            List<String> valueList = Collections.list(headerValues);
+            if (!CollectionUtil.isEmpty(valueList)) {
+                String headerValue = valueList.toString();
+                headersList.add(headerName + "=" + headerValue);
+            }
+        });
+
+        if (!headersList.isEmpty()) {
+            String tagValue = headersList.stream().collect(Collectors.joining("\n"));
+            tagValue = SpringMVCPluginConfig.Plugin.Http.HTTP_HEADERS_LENGTH_THRESHOLD > 0 ?
```

Comment:
`SpringMVCPluginConfig.Plugin.Http.HTTP_HEADERS_LENGTH_THRESHOLD > 0`, according to the comments, if the `value < 0`, then collect all, I don't find the codes.

Reply:
sorry, can't got your point, is the logic wrong ?

Reply:
According to your comments, `use negative values to keep and send the complete headers.`, where is the logic of that? Please consider to correct the comments or logic :) They should be matched.

Reply:
code logic is
```
   tagValue = SpringMVCPluginConfig.Plugin.Http.HTTP_HEADERS_LENGTH_THRESHOLD > 0 ?
                    StringUtil.cut(tagValue, SpringMVCPluginConfig.Plugin.Http.HTTP_HEADERS_LENGTH_THRESHOLD) : tagValue;
```
also changed the comments

Reply:
please help to check again thanks

Reply:
My mistake, ignore this.